src/stylefunction.js

```diff
@@ -1402,6 +1402,16 @@ export function stylefunction(
               ),
             ),
           );
+          const keepUpright = getValue(
+            layer,
+            'layout',
+            'text-keep-upright',
+            zoom,
+            f,
+            functionCache,
+            featureState,
+          );
+          text.setKeepUpright(keepUpright);
```

Comment:
For compatibility with older OpenLayers versions, this should be wrapped in an if-block:
```js
if (typeof text.setKeepUpright === 'function') {
  // ...
}